In [47]:
import matplotlib.pyplot as plt
import math
import pandas as pd
import numpy as np
import plotly.express as px 
import plotly.graph_objects as go
import plotly
from decimal import * #gives us perfect precision decimal points

# Reverse Process
Rather than reverse and add, this notebook demonstrates reverse and subtract.

In [77]:
pal = 196
n = 0
n_final = 1000

# generator, break condition removed as we can save computation time
def lychrel_generator(start) :
    a = start
    b = abs(a - int(str(a)[::-1]))
    while True :
        yield a
        a , b = b , abs(b - int(str(b)[::-1]))

lychrel_gen = lychrel_generator(pal)


title_text = "Frequency Of Digits for " + str(n_final) + " Steps of the Lychrel Process on 196" 
image_title = "images/196DigitDistributionForN" + str(n_final)

In [108]:
data = pd.read_csv('datasets/LychrelTo100000Integers.txt',  header=None)
data = data.values.tolist()[0][1:]
print(data)


[196, 295, 394, 493, 592, 689, 691, 788, 790, 879, 887, 978, 986, 1495, 1497, 1585, 1587, 1675, 1677, 1765, 1767, 1855, 1857, 1945, 1947, 1997, 2494, 2496, 2584, 2586, 2674, 2676, 2764, 2766, 2854, 2856, 2944, 2946, 2996, 3493, 3495, 3583, 3585, 3673, 3675, 3763, 3765, 3853, 3855, 3943, 3945, 3995, 4079, 4169, 4259, 4349, 4439, 4492, 4494, 4529, 4582, 4584, 4619, 4672, 4674, 4709, 4762, 4764, 4799, 4852, 4854, 4889, 4942, 4944, 4979, 5078, 5168, 5258, 5348, 5438, 5491, 5493, 5528, 5581, 5583, 5618, 5671, 5673, 5708, 5761, 5763, 5798, 5851, 5853, 5888, 5941, 5943, 5978, 5993, 6077, 6167, 6257, 6347, 6437, 6490, 6492, 6527, 6580, 6582, 6617, 6670, 6672, 6707, 6760, 6762, 6797, 6850, 6852, 6887, 6940, 6942, 6977, 6992, 7059, 7076, 7149, 7166, 7239, 7256, 7329, 7346, 7419, 7436, 7491, 7509, 7526, 7581, 7599, 7616, 7671, 7689, 7706, 7761, 7779, 7796, 7851, 7869, 7886, 7941, 7959, 7976, 7991, 8058, 8075, 8079, 8089, 8148, 8165, 8169, 8179, 8238, 8255, 8259, 8269, 8328, 8345, 8349, 8359, 8418

In [109]:

valid = True

worked = 0
didnt_work = 0

for lychrel in data:
    lychrel_gen = lychrel_generator(lychrel)
    i=0
    for num in lychrel_gen:
        if (i > n_final or num == int(str(num)[::-1])):
            found_text = "Palindrome found for " + str(lychrel) + " at  " + str(num) + " after " + str(i) + " steps."
            print(found_text if num == int(str(num)[::-1]) else "limit exceeded for " + str(lychrel))
            valid = num == int(str(num)[::-1])
            if valid:
                worked+=1
            else:
                didnt_work+=1
            break
        else:
            #print(i, num)
            i+=1
print(valid)
print("The proportion of success is", str(worked/(worked+didnt_work) ))

Palindrome found for 196 at  99 after 2 steps.
Palindrome found for 295 at  99 after 3 steps.
Palindrome found for 394 at  99 after 1 steps.
Palindrome found for 493 at  99 after 1 steps.
Palindrome found for 592 at  99 after 3 steps.
Palindrome found for 689 at  99 after 3 steps.
Palindrome found for 691 at  99 after 2 steps.
Palindrome found for 788 at  99 after 1 steps.
Palindrome found for 790 at  99 after 4 steps.
Palindrome found for 879 at  99 after 1 steps.
Palindrome found for 887 at  99 after 1 steps.
Palindrome found for 978 at  99 after 1 steps.
Palindrome found for 986 at  99 after 3 steps.
Palindrome found for 1495 at  999 after 6 steps.
Palindrome found for 1497 at  999 after 6 steps.
Palindrome found for 1585 at  999 after 11 steps.
Palindrome found for 1587 at  999 after 8 steps.
Palindrome found for 1675 at  909 after 4 steps.
Palindrome found for 1677 at  999 after 7 steps.
Palindrome found for 1765 at  99 after 7 steps.
Palindrome found for 1767 at  909 after 5 step

limit exceeded for 12965
Palindrome found for 13087 at  9999 after 6 steps.
limit exceeded for 13097
limit exceeded for 13197
Palindrome found for 13287 at  9999 after 6 steps.
Palindrome found for 13383 at  9999 after 3 steps.
limit exceeded for 13392
limit exceeded for 13393
limit exceeded for 13397
Palindrome found for 13508 at  9999 after 6 steps.
limit exceeded for 13523
Palindrome found for 13533 at  9999 after 5 steps.
Palindrome found for 13547 at  9999 after 7 steps.
limit exceeded for 13553
limit exceeded for 13555
limit exceeded for 13596
limit exceeded for 13597
Palindrome found for 13608 at  9999 after 6 steps.
limit exceeded for 13623
Palindrome found for 13633 at  9999 after 5 steps.
Palindrome found for 13638 at  9999 after 4 steps.
Palindrome found for 13667 at  9999 after 8 steps.
limit exceeded for 13694
limit exceeded for 13705
limit exceeded for 13716
Palindrome found for 13718 at  9009 after 6 steps.
limit exceeded for 13753
limit exceeded for 13755
Palindrome fou

Palindrome found for 17787 at  9999 after 7 steps.
limit exceeded for 17807
Palindrome found for 17813 at  9999 after 8 steps.
limit exceeded for 17838
Palindrome found for 17848 at  9999 after 6 steps.
Palindrome found for 17867 at  9009 after 5 steps.
Palindrome found for 17897 at  9999 after 9 steps.
limit exceeded for 17907
Palindrome found for 17913 at  9999 after 8 steps.
Palindrome found for 17915 at  9999 after 11 steps.
limit exceeded for 17957
limit exceeded for 17965
limit exceeded for 17989
limit exceeded for 18037
limit exceeded for 18047
limit exceeded for 18056
Palindrome found for 18057 at  9999 after 5 steps.
limit exceeded for 18096
Palindrome found for 18097 at  9999 after 7 steps.
limit exceeded for 18147
Palindrome found for 18157 at  9999 after 5 steps.
Palindrome found for 18180 at  9999 after 1 steps.
Palindrome found for 18194 at  9999 after 10 steps.
limit exceeded for 18237
Palindrome found for 18257 at  9999 after 5 steps.
limit exceeded for 18267
Palindrome

limit exceeded for 23607
Palindrome found for 23622 at  99 after 6 steps.
Palindrome found for 23637 at  9999 after 2 steps.
Palindrome found for 23666 at  9999 after 11 steps.
limit exceeded for 23693
Palindrome found for 23704 at  9999 after 12 steps.
limit exceeded for 23715
limit exceeded for 23717
limit exceeded for 23752
limit exceeded for 23754
limit exceeded for 23756
Palindrome found for 23757 at  5445 after 3 steps.
limit exceeded for 23782
Palindrome found for 23786 at  9999 after 6 steps.
limit exceeded for 23812
Palindrome found for 23846 at  9009 after 4 steps.
limit exceeded for 23852
limit exceeded for 23877
Palindrome found for 23887 at  54945 after 1 steps.
limit exceeded for 23902
Palindrome found for 23934 at  9999 after 5 steps.
Palindrome found for 23938 at  9999 after 2 steps.
Palindrome found for 23946 at  9009 after 4 steps.
limit exceeded for 23952
limit exceeded for 23954
Palindrome found for 23996 at  9009 after 6 steps.
Palindrome found for 24076 at  9999 a

limit exceeded for 30746
Palindrome found for 30781 at  9999 after 6 steps.
limit exceeded for 30783
Palindrome found for 30785 at  9009 after 3 steps.
Palindrome found for 30786 at  9999 after 12 steps.
Palindrome found for 30841 at  9999 after 7 steps.
Palindrome found for 30849 at  9009 after 6 steps.
limit exceeded for 30875
Palindrome found for 30881 at  9999 after 6 steps.
limit exceeded for 30889
Palindrome found for 30929 at  9999 after 9 steps.
Palindrome found for 30931 at  9999 after 12 steps.
Palindrome found for 30959 at  9999 after 6 steps.
Palindrome found for 30961 at  9999 after 8 steps.
Palindrome found for 30963 at  5445 after 2 steps.
Palindrome found for 30967 at  9009 after 6 steps.
limit exceeded for 30975
Palindrome found for 30979 at  9999 after 11 steps.
Palindrome found for 30981 at  9999 after 6 steps.
limit exceeded for 30983
limit exceeded for 31079
limit exceeded for 31348
limit exceeded for 31379
Palindrome found for 31389 at  9999 after 11 steps.
Palind

limit exceeded for 35933
Palindrome found for 35969 at  9999 after 7 steps.
limit exceeded for 35975
limit exceeded for 35983
Palindrome found for 35999 at  9009 after 6 steps.
limit exceeded for 36029
limit exceeded for 36055
Palindrome found for 36065 at  9999 after 5 steps.
limit exceeded for 36074
limit exceeded for 36075
Palindrome found for 36165 at  9999 after 5 steps.
limit exceeded for 36175
limit exceeded for 36255
limit exceeded for 36275
limit exceeded for 36285
limit exceeded for 36329
Palindrome found for 36339 at  9999 after 5 steps.
limit exceeded for 36351
Palindrome found for 36360 at  9999 after 3 steps.
Palindrome found for 36361 at  9999 after 5 steps.
Palindrome found for 36365 at  9999 after 5 steps.
limit exceeded for 36375
Palindrome found for 36381 at  9009 after 5 steps.
Palindrome found for 36388 at  5445 after 3 steps.
limit exceeded for 36419
limit exceeded for 36429
Palindrome found for 36469 at  9999 after 2 steps.
Palindrome found for 36488 at  5445 aft

limit exceeded for 40574
Palindrome found for 40580 at  9999 after 6 steps.
Palindrome found for 40582 at  9999 after 6 steps.
limit exceeded for 40635
limit exceeded for 40650
Palindrome found for 40660 at  9999 after 11 steps.
limit exceeded for 40665
limit exceeded for 40694
limit exceeded for 40712
Palindrome found for 40725 at  9999 after 11 steps.
Palindrome found for 40732 at  9999 after 12 steps.
limit exceeded for 40743
Palindrome found for 40745 at  9009 after 4 steps.
Palindrome found for 40780 at  9999 after 6 steps.
Palindrome found for 40782 at  9999 after 6 steps.
limit exceeded for 40784
limit exceeded for 40785
Palindrome found for 40840 at  9009 after 4 steps.
limit exceeded for 40848
limit exceeded for 40874
Palindrome found for 40880 at  9999 after 6 steps.
Palindrome found for 40888 at  9999 after 9 steps.
limit exceeded for 40928
limit exceeded for 40930
Palindrome found for 40958 at  9999 after 6 steps.
Palindrome found for 40960 at  9999 after 11 steps.
Palindro

limit exceeded for 45974
Palindrome found for 45982 at  9999 after 12 steps.
limit exceeded for 45998
limit exceeded for 46009
Palindrome found for 46028 at  9009 after 4 steps.
Palindrome found for 46054 at  99 after 6 steps.
Palindrome found for 46073 at  9009 after 1 steps.
Palindrome found for 46074 at  99 after 6 steps.
limit exceeded for 46099
Palindrome found for 46174 at  99 after 6 steps.
Palindrome found for 46254 at  99 after 6 steps.
Palindrome found for 46274 at  99 after 6 steps.
limit exceeded for 46284
Palindrome found for 46328 at  9009 after 4 steps.
limit exceeded for 46338
Palindrome found for 46350 at  9009 after 4 steps.
Palindrome found for 46360 at  9999 after 4 steps.
Palindrome found for 46374 at  99 after 6 steps.
limit exceeded for 46380
Palindrome found for 46387 at  9009 after 7 steps.
limit exceeded for 46418
Palindrome found for 46428 at  9009 after 4 steps.
limit exceeded for 46459
Palindrome found for 46468 at  9999 after 4 steps.
limit exceeded for 46

limit exceeded for 50742
limit exceeded for 50744
Palindrome found for 50781 at  9999 after 6 steps.
Palindrome found for 50783 at  9999 after 6 steps.
limit exceeded for 50784
Palindrome found for 50847 at  9999 after 10 steps.
Palindrome found for 50873 at  9009 after 6 steps.
Palindrome found for 50887 at  9009 after 3 steps.
limit exceeded for 50927
Palindrome found for 50957 at  9999 after 3 steps.
Palindrome found for 50961 at  9999 after 11 steps.
Palindrome found for 50965 at  5445 after 2 steps.
Palindrome found for 50973 at  9009 after 6 steps.
limit exceeded for 50977
Palindrome found for 50978 at  9009 after 5 steps.
Palindrome found for 50981 at  9999 after 6 steps.
limit exceeded for 51058
limit exceeded for 51077
limit exceeded for 51346
limit exceeded for 51377
limit exceeded for 51387
Palindrome found for 51418 at  9999 after 3 steps.
limit exceeded for 51427
Palindrome found for 51438 at  9009 after 7 steps.
limit exceeded for 51446
Palindrome found for 51467 at  9999

Palindrome found for 60960 at  9009 after 2 steps.
Palindrome found for 60964 at  9999 after 8 steps.
limit exceeded for 60972
limit exceeded for 60976
limit exceeded for 60977
limit exceeded for 60980
Palindrome found for 61057 at  9009 after 4 steps.
Palindrome found for 61076 at  5445 after 2 steps.
limit exceeded for 61149
limit exceeded for 61159
limit exceeded for 61279
limit exceeded for 61345
Palindrome found for 61376 at  5445 after 2 steps.
limit exceeded for 61386
Palindrome found for 61417 at  9999 after 1 steps.
Palindrome found for 61426 at  99 after 6 steps.
Palindrome found for 61437 at  9999 after 11 steps.
limit exceeded for 61445
limit exceeded for 61466
Palindrome found for 61476 at  5445 after 2 steps.
limit exceeded for 61479
limit exceeded for 61488
limit exceeded for 61523
Palindrome found for 61558 at  9999 after 10 steps.
limit exceeded for 61562
Palindrome found for 61568 at  9999 after 3 steps.
Palindrome found for 61570 at  9009 after 2 steps.
limit exceede

limit exceeded for 65382
limit exceeded for 65395
Palindrome found for 65405 at  9999 after 4 steps.
limit exceeded for 65426
limit exceeded for 65436
limit exceeded for 65439
limit exceeded for 65448
limit exceeded for 65467
limit exceeded for 65476
limit exceeded for 65487
limit exceeded for 65495
Palindrome found for 65518 at  9999 after 7 steps.
Palindrome found for 65522 at  9999 after 11 steps.
Palindrome found for 65528 at  9999 after 12 steps.
Palindrome found for 65530 at  9999 after 9 steps.
limit exceeded for 65571
limit exceeded for 65572
Palindrome found for 65592 at  9009 after 4 steps.
limit exceeded for 65609
limit exceeded for 65613
limit exceeded for 65639
Palindrome found for 65642 at  9009 after 4 steps.
Palindrome found for 65683 at  9009 after 3 steps.
limit exceeded for 65699
Palindrome found for 65730 at  9999 after 9 steps.
limit exceeded for 65732
Palindrome found for 65733 at  9009 after 7 steps.
limit exceeded for 65762
Palindrome found for 65822 at  9999 af

Palindrome found for 69722 at  9999 after 8 steps.
Palindrome found for 69762 at  9999 after 11 steps.
Palindrome found for 69782 at  9009 after 4 steps.
Palindrome found for 69783 at  9999 after 10 steps.
Palindrome found for 69792 at  9999 after 4 steps.
Palindrome found for 69813 at  9999 after 12 steps.
Palindrome found for 69823 at  9009 after 5 steps.
Palindrome found for 69842 at  9999 after 6 steps.
limit exceeded for 69846
limit exceeded for 69872
Palindrome found for 69882 at  9009 after 4 steps.
Palindrome found for 69886 at  99 after 6 steps.
Palindrome found for 69895 at  9999 after 1 steps.
limit exceeded for 69926
Palindrome found for 69932 at  9009 after 6 steps.
Palindrome found for 69940 at  9999 after 6 steps.
limit exceeded for 69956
Palindrome found for 69964 at  9009 after 7 steps.
limit exceeded for 69976
limit exceeded for 69977
Palindrome found for 69993 at  9999 after 3 steps.
Palindrome found for 70066 at  5445 after 2 steps.
limit exceeded for 70069
Palindro

Palindrome found for 75905 at  9999 after 3 steps.
limit exceeded for 75913
Palindrome found for 75921 at  9999 after 8 steps.
Palindrome found for 75925 at  9009 after 7 steps.
limit exceeded for 75926
limit exceeded for 75965
limit exceeded for 75971
Palindrome found for 75989 at  9009 after 7 steps.
Palindrome found for 75995 at  9999 after 7 steps.
limit exceeded for 76006
Palindrome found for 76009 at  9999 after 8 steps.
Palindrome found for 76025 at  9999 after 10 steps.
Palindrome found for 76051 at  9999 after 7 steps.
Palindrome found for 76061 at  9999 after 2 steps.
Palindrome found for 76070 at  9999 after 8 steps.
Palindrome found for 76071 at  9009 after 5 steps.
limit exceeded for 76096
Palindrome found for 76099 at  9009 after 7 steps.
Palindrome found for 76108 at  5445 after 2 steps.
Palindrome found for 76161 at  9999 after 2 steps.
Palindrome found for 76171 at  9009 after 5 steps.
Palindrome found for 76188 at  9999 after 11 steps.
limit exceeded for 76198
limit e

limit exceeded for 80279
limit exceeded for 80287
limit exceeded for 80289
Palindrome found for 80299 at  9009 after 4 steps.
Palindrome found for 80349 at  9009 after 4 steps.
Palindrome found for 80353 at  9009 after 2 steps.
limit exceeded for 80358
limit exceeded for 80369
limit exceeded for 80378
Palindrome found for 80384 at  9999 after 6 steps.
Palindrome found for 80394 at  9009 after 6 steps.
limit exceeded for 80425
limit exceeded for 80434
limit exceeded for 80439
limit exceeded for 80445
Palindrome found for 80453 at  9009 after 2 steps.
limit exceeded for 80474
Palindrome found for 80484 at  9999 after 6 steps.
limit exceeded for 80487
limit exceeded for 80488
limit exceeded for 80489
limit exceeded for 80496
Palindrome found for 80529 at  9999 after 11 steps.
Palindrome found for 80531 at  9999 after 6 steps.
limit exceeded for 80539
limit exceeded for 80558
Palindrome found for 80566 at  9999 after 8 steps.
limit exceeded for 80569
limit exceeded for 80570
Palindrome fou

limit exceeded for 83945
limit exceeded for 83984
limit exceeded for 83990
limit exceeded for 84009
limit exceeded for 84019
Palindrome found for 84025 at  9009 after 7 steps.
limit exceeded for 84028
limit exceeded for 84029
Palindrome found for 84044 at  9999 after 4 steps.
Palindrome found for 84070 at  54945 after 2 steps.
limit exceeded for 84080
limit exceeded for 84090
Palindrome found for 84099 at  9999 after 4 steps.
limit exceeded for 84109
limit exceeded for 84117
limit exceeded for 84119
Palindrome found for 84127 at  9999 after 11 steps.
limit exceeded for 84129
limit exceeded for 84159
limit exceeded for 84179
limit exceeded for 84180
limit exceeded for 84190
Palindrome found for 84199 at  9999 after 4 steps.
limit exceeded for 84229
Palindrome found for 84239 at  9009 after 1 steps.
Palindrome found for 84247 at  9999 after 1 steps.
Palindrome found for 84249 at  9999 after 1 steps.
limit exceeded for 84259
Palindrome found for 84270 at  54945 after 2 steps.
Palindrome f

limit exceeded for 89109
limit exceeded for 89129
Palindrome found for 89130 at  9999 after 10 steps.
Palindrome found for 89140 at  9999 after 3 steps.
limit exceeded for 89149
Palindrome found for 89157 at  9009 after 4 steps.
limit exceeded for 89159
limit exceeded for 89167
limit exceeded for 89169
limit exceeded for 89209
Palindrome found for 89220 at  9009 after 7 steps.
Palindrome found for 89239 at  5445 after 2 steps.
Palindrome found for 89240 at  9999 after 3 steps.
limit exceeded for 89250
limit exceeded for 89269
Palindrome found for 89270 at  9009 after 3 steps.
limit exceeded for 89279
limit exceeded for 89287
Palindrome found for 89289 at  9009 after 1 steps.
Palindrome found for 89296 at  9999 after 5 steps.
Palindrome found for 89297 at  9999 after 1 steps.
Palindrome found for 89304 at  9999 after 7 steps.
Palindrome found for 89330 at  9999 after 10 steps.
Palindrome found for 89340 at  9999 after 3 steps.
limit exceeded for 89349
limit exceeded for 89353
limit exce

limit exceeded for 92469
limit exceeded for 92475
limit exceeded for 92479
limit exceeded for 92494
Palindrome found for 92508 at  9999 after 11 steps.
Palindrome found for 92510 at  9009 after 4 steps.
limit exceeded for 92518
limit exceeded for 92537
limit exceeded for 92545
limit exceeded for 92548
limit exceeded for 92555
limit exceeded for 92559
limit exceeded for 92568
limit exceeded for 92569
limit exceeded for 92598
Palindrome found for 92610 at  9009 after 4 steps.
limit exceeded for 92618
Palindrome found for 92628 at  9999 after 1 steps.
limit exceeded for 92636
limit exceeded for 92640
Palindrome found for 92647 at  9009 after 5 steps.
limit exceeded for 92648
limit exceeded for 92666
limit exceeded for 92668
limit exceeded for 92700
Palindrome found for 92720 at  9999 after 5 steps.
Palindrome found for 92728 at  9999 after 1 steps.
limit exceeded for 92748
limit exceeded for 92758
Palindrome found for 92760 at  9009 after 6 steps.
Palindrome found for 92789 at  5445 after

limit exceeded for 96108
limit exceeded for 96138
limit exceeded for 96158
Palindrome found for 96178 at  9009 after 1 steps.
limit exceeded for 96186
limit exceeded for 96188
Palindrome found for 96196 at  9009 after 3 steps.
limit exceeded for 96198
limit exceeded for 96208
Palindrome found for 96218 at  9999 after 4 steps.
limit exceeded for 96219
limit exceeded for 96226
Palindrome found for 96228 at  9009 after 4 steps.
limit exceeded for 96238
Palindrome found for 96268 at  9999 after 1 steps.
Palindrome found for 96279 at  99 after 6 steps.
limit exceeded for 96298
limit exceeded for 96308
Palindrome found for 96317 at  9999 after 3 steps.
limit exceeded for 96319
Palindrome found for 96323 at  9009 after 6 steps.
Palindrome found for 96333 at  9999 after 8 steps.
limit exceeded for 96349
Palindrome found for 96378 at  9009 after 1 steps.
Palindrome found for 96382 at  9009 after 6 steps.
Palindrome found for 96387 at  9009 after 5 steps.
limit exceeded for 96398
Palindrome foun

limit exceeded for 98658
Palindrome found for 98666 at  9009 after 7 steps.
limit exceeded for 98677
limit exceeded for 98678
limit exceeded for 98696
Palindrome found for 98698 at  9009 after 1 steps.
Palindrome found for 98700 at  9999 after 11 steps.
Palindrome found for 98729 at  5445 after 2 steps.
limit exceeded for 98758
limit exceeded for 98778
Palindrome found for 98788 at  9999 after 1 steps.
Palindrome found for 98790 at  9999 after 11 steps.
limit exceeded for 98803
Palindrome found for 98820 at  9009 after 4 steps.
Palindrome found for 98830 at  9999 after 4 steps.
Palindrome found for 98853 at  9999 after 8 steps.
limit exceeded for 98858
limit exceeded for 98877
Palindrome found for 98893 at  9009 after 5 steps.
Palindrome found for 98933 at  9999 after 6 steps.
Palindrome found for 98957 at  9009 after 7 steps.
Palindrome found for 98963 at  9999 after 9 steps.
Palindrome found for 98968 at  9999 after 11 steps.
limit exceeded for 98971
Palindrome found for 98983 at  99

For the sample size of 10000 integers the success rate of this method is 0.947, yet by increasing it to 100000 the success rate drops to 0.0.540531561461794. It is interesting to note that of those that do work, almost all of them seem to converge to a number of all 9s (99, 999, 9999, etc) or 0s bookended by 9s (909, 9009, 90009, etc). There also seem to be a handful of instances of numbers converging on 5445 or 54945.

It is also interesting to note that this method works for every single 3 digit lychrel.

Interestingly, it would appear that for all that dont work fall into the same thread of continually alternating between 2178 and 6534, as by the reverse process each of these numbers subtracts into the other. I have not yet investigated in a more rigorous fashion, but i am yet to find a counterexample.